### ISSUE TITLE
New Forum Relationship to Github Issues 

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#1519]
I think this is worth discussing so we can pretty clear about goals and responsibilities. EG: what should be discussed where? 

As rough guidelines, I have in mind:

Github issues:
- bugs 
- task tracking
- feature requests (once accepted we close and add a link in the feature request list)
- technical discussions (not involving broader community)

Website:
- help documentation
- how to anything
- rules

Forum:
- open ended discussions
- map threads (this worked pretty well in Source forge. There has not been a mass migration to github maps project. TBD exactly how we get folks to upload their maps early in the map making process)
- game play and strategy topics
- help link that points people to the website help page
- PBEM requests (this is something I'd like to have built into the game)
- tourney's?  we can do quite a bit with by-hand tracking. this is also something I'd like to build into the game.
- map maker to map maker communication

Okay, for discussion, how does the above list feel? The "discussion" bullet point is most interesting. Technical discussions are probably best on github issues, but the more open ended ones involving the broader community are probably better on forum. That is actually more or less how it was when SF was an official forum and we were using github issues.

**Comment by prastle** [comment]
Need PBEM 

need a ladder

rest looks good
but how to add old ladder?
prob imposs

thus statistics of old important if it can be retrieved
and start fresh 
a nwo ladder a revised aglobal

and an ancient aa50 for me :)
lol

**Comment by prastle** [comment]
also if a simple clone can be built a ladder of any version can be easily added

**Comment by DanVanAtta** [comment]
"PBEM requests"  is listed under forum

Ladder is missing intentionally. I think we can get one built into the game in the next 3 to 6 months. Maybe we might want to do something else in the short term, but if we migrate again anyways that soon, I think we are best off waiting.

**Comment by prastle** [comment]
ah maybe you fail to understand but we havent had a ladder in 2 years

this is peeps that love the game that are left

ldder broken long time

to be blunt not impolite

you developers focused on maps

what you failed to understand was 0 players maps no matter

wwant a site
fix ladder etc
then add maps
and neat things

**Comment by DanVanAtta** [comment]
The ladder never drew me in. I've probably undervalued that priority. 

Maps is the most visible focus. I think in part we are here because there has been a lot to do. The focus has actually been on admin/dev productivity and overhead:
- first migrated to github, allowed us to update the code base more rapidly, and allows multiple people to work easily on the code at the same time. Allowed greater bandwidth for code updates.
- fixed the deployment pipeline. It used to be an overall two week effort, with a days worth of things to update and track
- eased the pain of re-installing the game. Cut the download size of engine and maps lower (mostly by converting sounds to mp3) and launched the github.io  website to make  the download steps and instructions easy to do.
- map migration was part of the github migration, to get out of source forge. In the process we removed most map admin tasks to reduce overhead.

Once all that was done, i was planning on focusing on gameplay aspects : )

The new forum + new website, I think we are set on the site. Game challenges and ladder is coming up quick on the list.. we're getting there..

**Comment by DanVanAtta** [comment]
So, we made it so we can change the code easily. But changing the code easily requires new installs, so we made it so we don't lose 10% of the player base each time a new version comes out (hopefully... 1.9 migration was bumpy).

yeah, maps should have been after fixing up the lobby side. We could have stayed in source forge longer for maps,  and instead we could have focused on game challenges / lobby.  At least though we have fully migrated all our code / assets to github now

**Comment by panther2** [comment]
Being not familiar with the term "ladder" in the context of gaming, could someone please explain what a "ladder" actually is?

**Comment by prastle** [comment]
easier if you just look here since ours cant be viewed anymore

http://www.casesladder.com/

**Comment by panther2** [comment]
Ah, I see, sort of a ranking system... thanks!

**Comment by prastle** [comment]
Also @DanVanAtta 

I think what you have accomplished with the developers is amazing! compared to old.
In game downloads etc etc etc etc

But we need to now concentrate on what peeps used to come to triplea for.

A community that is centralized with a competition module.

**Comment by DanVanAtta** [comment]
Follow up tasks:  post this information to the new lobby in a 'where to post'

**Comment by FrostionAAA** [comment]
Joining this discussion a bit late, as usual ;-) But, as we can see, the new forum is ready to host map maker discussions under the category ”_Map Making - Discuss the technicalities of map making and XML editing._” Another forum category is “_Maps & Mods - Help out, share ideas and work on specific Maps and Mods._”


**Map Making**
I would think it would be a good idea to keep all guides and help also relating to the process of map releases/publishing/uploading to Gighub under the Map Making category of the new forum, maybe as sticky post under Map Making. This forum category should become the place potential mapmakers (who do not necessarily have any idea about what Github is or even know of its existence) would hang out, read and learn about map making while working on their maps.

Am I correct that this here; https://github.com/triplea-maps/Project, is the current and only guide we have about how to publish? As I see it would make good sense to keep the map making process and the publishing guide at one place. The Github guide should move to the forum. Then this guide could of course, as time passes and map makers respond to the post with questions, be perfected to become an  idiot proof step-by-step guide that would make even the most Github-frightened mapmaker get his map released.

**Maps & Mods**
I really hope that this category could become the place where map makers could announce their maps, as well as a place where players could post spontaneously about maps. I would guess some maps/mapmakers would have map-dedicated threads going, but the doors should be open to whatever people have on their mind about specific maps, maps in general or unmade maps. In this context I would like to ask a question: Should this also be the place to report map errors? The line is thin when it comes to de difference between flaws in maps, map setup, wishing a change in map rules etc.
Or should players, maybe via a sticky thread under the Maps & Mods category be encouraged to post their “issues” under https://github.com/triplea-maps/Project/issues ? If this is the case, then there should be made a post at the forum. Maybe “How to report a map bug” sticky post, something that could be understood by even the players who do not have a Github account but would be instructed to make one if serious about the bug report. I would think some players would just post about the bug at the forum, maybe discussing it, and this door should be open and encouraged.

**Anyway**, what do people think about map publishing guide at forum? And how should bug reporting be done? A bug report guide pointing to Github?

**PS:** I ran into the heated discussion about if/what old forum posts should/could be copied over to the new forum. I agree that there is an overwhelming amount of outdated info and posts that would be meaningless to transfer to the new forum, but as Cernel pointed out, when it comes to map making related posts, these are pretty useful to potential mapmakers. They are essentially years of work, experience and ideas that could help new map makers in their map making efforts. What is the plan? Is it possible to have an “Archives” section under Map Making at the new forum? Maybe something that would only be accessible via the new forum? Or a link to the old forum? Or should everybody just be encouraged to copy/past content over by themselves when relevant?

**Comment by prastle** [comment]
@FrostionAAA  in one of the many threads it was decided (i forget which one) that all are responsible for importing what they want.  No massive data transfer is going to happen.  All of your above points make sense to me.  Thus I tried to transfer things that I felt were important from war club.  Hopefully the same can be done for Source Forge.  Hopefully it can be archived as read only somehow.  I did ask that question elsewhere.  No one has answered it yet to my knowledge.
